xline/src/server/xline_server.rs

```diff
@@ -285,124 +238,133 @@ where
-                }
-            }
-        }
-    }
-
     /// Init `KvServer`, `LockServer`, `LeaseServer`, `WatchServer` and `CurpServer`
     /// for the Xline Server.
     #[allow(clippy::type_complexity)] // it is easy to read
-    async fn init_servers(
+    async fn init_servers<S: StorageApi>(
         &self,
-    ) -> (
+        persistent: Arc<S>,
+        key_pair: Option<(EncodingKey, DecodingKey)>,
+    ) -> Result<(
         KvServer<S>,
         LockServer,
         Arc<LeaseServer<S>>,
         AuthServer<S>,
         WatchServer<S>,
         MaintenanceServer<S>,
-        CurpServer,
-    ) {
-        let curp_server = match self.storage_cfg {
+        CurpServer<S>,
+    )> {
+        let (header_gen, id_gen, auth_revision_gen) = Self::construct_generator(&self.cluster_info);
+        let lease_collection = Self::construct_lease_collection(
+            self.curp_cfg.heartbeat_interval,
+            self.curp_cfg.candidate_timeout_ticks,
+        );
+
+        let (kv_storage, lease_storage, auth_storage, watcher) =
+            Self::construct_underlying_storages(
+                self.is_leader,
+                &persistent,
+                lease_collection,
+                &header_gen,
+                &auth_revision_gen,
+                key_pair,
+            )?;
+        let client = Arc::new(
+            Client::<Command>::new(self.cluster_info.get_all_members(), self.client_timeout).await,
+        );
+        let index_barrier = Arc::new(IndexBarrier::new());
+        let id_barrier = Arc::new(IdBarrier::new());
+
+        let state = State::new(Arc::clone(&lease_storage));
+        let ce = CommandExecutor::new(
+            Arc::clone(&kv_storage),
+            Arc::clone(&auth_storage),
+            Arc::clone(&lease_storage),
+            Arc::clone(&persistent),
+            Arc::clone(&index_barrier),
+            Arc::clone(&id_barrier),
+            header_gen.revision_arc(),
+            Arc::clone(&auth_revision_gen),
+        );
+
+        Ok((
+            KvServer::new(
+                Arc::clone(&kv_storage),
+                Arc::clone(&auth_storage),
+                Arc::clone(&index_barrier),
+                Arc::clone(&id_barrier),
+                self.range_retry_timeout,
+                Arc::clone(&client),
+                self.cluster_info.get_self_id().clone(),
+            ),
+            LockServer::new(
+                Arc::clone(&client),
+                Arc::clone(&id_gen),
+                self.cluster_info.get_self_id().clone(),
+                self.cluster_info.get_self_address().to_owned(),
+            ),
+            LeaseServer::new(
+                Arc::clone(&lease_storage),
+                Arc::clone(&auth_storage),
+                Arc::clone(&client),
+                Arc::clone(&id_gen),
+                Arc::clone(&self.cluster_info),
+            ),
+            AuthServer::new(
+                Arc::clone(&auth_storage),
+                Arc::clone(&client),
```

Comment:
Some of these clone are not necessary.

changed version:
```rust, no_run
            KvServer::new(
                kv_storage,
                Arc::clone(&auth_storage),
                index_barrier,
                id_barrier,
                self.range_retry_timeout,
                Arc::clone(&client),
                self.cluster_info.get_self_id().clone(),
            ),
            LockServer::new(
                Arc::clone(&client),
                Arc::clone(&id_gen),
                self.cluster_info.get_self_id().clone(),
                self.cluster_info.get_self_address().to_owned(),
            ),
            LeaseServer::new(
                lease_storage,
                Arc::clone(&auth_storage),
                Arc::clone(&client),
                id_gen,
                Arc::clone(&self.cluster_info),
            ),
            AuthServer::new(
                auth_storage,
                client,
                self.cluster_info.get_self_id().clone(),
            ),
            WatchServer::new(watcher, Arc::clone(&header_gen)),
            MaintenanceServer::new(persistent, header_gen),
```

Reply:
Done